pylint/checkers/variables.py

```diff
@@ -2210,34 +2210,31 @@ def _is_variable_violation(
                 and isinstance(defstmt.parent, nodes.If)
                 and in_type_checking_block(defstmt)
                 and not in_type_checking_block(node)
+                and not VariablesChecker._is_variable_annotation_in_function(node)
+                and not defstmt.parent.parent_of(node)
```

Comment:
pylint raises `too-many-boolean-expressions` here, but at the same time, I don't think it is a good idea to add another `if` branch just to bypass this warning.

Any suggestions? One workaround would be to create another method to perform a few of these checks.

Reply:
We may need to disable the message here. But first, do you know why `and isinstance(defstmt.parent, nodes.If)` is needed? That seems fishy, since `in_type_checking_block(defstmt)` is supposed to take care of that (and allow for defstmt.parent being something else, like a `with`).

Reply:
I guess it's because it's used as `outermost_type_checking_guard` below.

Reply:
If only... :D it would be nice if mypy knew and pre-commit warned us of that kind of thing.